### PR TITLE
[DE-475] agregado método para obtener fields id

### PR DESCRIPTION
**Pull request by Cromeror**

### CODE REVIEW

File: Doppler.HtmlEditorApi/DopplerHtmlDocument.cs [hunk 1]
```diff
@@ -51,6 +53,15 @@ public string GetDopplerContent()
     public string GetHeadContent()
         => _headNode?.InnerHtml;
 
+    public IEnumerable<int> GetFieldsId()
+    {
+        return FIELD_ID_TAG_REGEX.Matches(_contentNode.InnerHtml)
+        .Select(x => x.Groups[1].Value)
+        .Select(int.Parse)
+        .Distinct();
+    }
```
Comment (andresmoschini) [inline Doppler.HtmlEditorApi/DopplerHtmlDocument.cs:62]: What do you think about using the short version?

```suggestion
    public IEnumerable<int> GetFieldsId()
        => FIELD_ID_TAG_REGEX.Matches(_contentNode.InnerHtml)
            .Select(x => x.Groups[1].Value)
            .Select(int.Parse)
            .Distinct();
```

(I have no a strong opinion on this)

File: Doppler.HtmlEditorApi.Test/DopplerHtmlDocumentTest.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Net;
+using System.Threading.Tasks;
+using Doppler.HtmlEditorApi.Test.Utils;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using System.Text.Json;
+using Xunit.Abstractions;
```
Comment (andresmoschini) [inline Doppler.HtmlEditorApi.Test/DopplerHtmlDocumentTest.cs:12]: What do you think about removing unused dependencies?

```suggestion
using Xunit.Abstractions;
```

File: Doppler.HtmlEditorApi/DopplerHtmlDocument.cs [hunk 3]
```diff
@@ -51,6 +53,14 @@ public string GetDopplerContent()
...
 
+    public IEnumerable<int> GetFieldsId()
+    {
+        return FIELD_ID_TAG_REGEX.Matches(_contentNode.InnerHtml)
+        .Select(x => int.Parse(x.Groups[1].ValueSpan))
+        .Distinct();
+    }
+
+
     public void ReplaceFieldNameTagsByFieldIdTags(Func<string, int?> getFieldIdOrNullFunc)
```
Comment (andresmoschini) [inline Doppler.HtmlEditorApi/DopplerHtmlDocument.cs:64]:
```suggestion
    }

    public void ReplaceFieldNameTagsByFieldIdTags(Func<string, int?> getFieldIdOrNullFunc)
```

File: Doppler.HtmlEditorApi.Test/DopplerHtmlDocumentTest.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+{
+    private readonly ITestOutputHelper _output;
+
+    public DopplerHtmlDocumentTest(ITestOutputHelper output)
+    {
+        _output = output;
+    }
+
+    [Fact]
+    public void get_distinct_fields_id()
```
Comment (andresmoschini) [inline Doppler.HtmlEditorApi.Test/DopplerHtmlDocumentTest.cs:28]: What do you think about this name?

```suggestion
    public void GetFieldsId_should_return_distinct_field_ids()
```

Comment (andresmoschini) [inline Doppler.HtmlEditorApi.Test/DopplerHtmlDocumentTest.cs:28]: What do you think about adding a test scenario for when there are no ids?

Comment (Cromeror) [reply Doppler.HtmlEditorApi.Test/DopplerHtmlDocumentTest.cs:28]: What should happen if you don't have ids?

Comment (andresmoschini) [reply Doppler.HtmlEditorApi.Test/DopplerHtmlDocumentTest.cs:28]: I think that `Assert.Equal(0, result.Count());` would be fine

**Review by maurocardinali — COMMENTED** [review]
🚀

**Comment by andresmoschini** [comment]
:shipit:

**Pull request merged**